app/controllers/state_registrants_controller.rb

```diff
@@ -21,7 +21,7 @@ def update
       go_to_paper and return
     end
     if !@registrant.eligible?
-      redirect_to(registrant_ineligible_url(@registrant)) and return
+      @registrant.skip_state_flow!
```

Comment:
I think you just want to have both lines here and that should do it?

@registrant.skip_state_flow!
redirect_to(registrant_ineligible_url(@registrant)) and return